### PR TITLE
[scutil] Fix a couple of minor compilation warnings; NFCI

### PR DESCRIPTION
**Pull request by bryanpkc**
- Do not use `STDC FENV_ACCESS` pragma on AArch64 which is not supported by GCC and Clang.
- Use unsigned wrap function for `DBLUINT64`.

### CODE REVIEW

File: lib/scutil/host-fp-folding.c [hunk 1]
```diff
@@ -55,12 +55,6 @@ fold_sanity_check(void)
   assert(sizeof(float128_t) == 16);
 }
 
-/*
- *  C99 feature: alert the compiler that the following code cares about
- *  the dynamic floating-point environment.
- */
-#pragma STDC FENV_ACCESS ON
```
Comment (shivaramaarao) [inline lib/scutil/host-fp-folding.c:62]: I understand that clang supports this pragma.  Is the pragma not important for the functionality of the code below?

Comment (bryanpkc) [reply lib/scutil/host-fp-folding.c:62]: LLVM 12 ignores this pragma when running without strict FP support. See https://reviews.llvm.org/D90316 for information.

Comment (bryanpkc) [reply lib/scutil/host-fp-folding.c:62]: It seems that `HasStrictFP` is `true` for X86 and PPC, but not for AArch64.

Comment (bryanpkc) [reply lib/scutil/host-fp-folding.c:62]: Fixed.

**Review by pawosm-arm — APPROVED** [review]

File: lib/scutil/host-fp-folding.c (hunk 1 above)
Comment (bryanpkc) [reply lib/scutil/host-fp-folding.c:62]: @shivaramaarao Please take a look at this again?

**Comment by bryanpkc** [comment]
ping @shivaramaarao @SouraVX

**Comment by bryanpkc** [comment]
ping @shivaramaarao

**Comment by bryanpkc** [comment]
ping @shivaramaarao

**Review by shivaramaarao — APPROVED** [review]

**Pull request merged**